src/pages/home/ReportScreen.js

```diff
@@ -165,7 +165,8 @@ class ReportScreen extends React.Component {
         // It possible that we may not have the report object yet in Onyx yet e.g. we navigated to a URL for an accessible report that
         // is not stored locally yet. If props.report.reportID exists, then the report has been stored locally and nothing more needs to be done.
         // If it doesn't exist, then we fetch the report from the API.
-        if (this.props.report.reportID) {
+        // If the client is offline, we cannot fetch the report, so let's just return.
```

Comment:
This comment explains what we are doing but not "why".

Looking at this - and I don't understand why being offline should prevent us from making this request. Here's my understanding of how our networking layer works:

1. `OpenReport` is a "write" request because we update the "last read" while fetching the report
2. If we are offline we will queue any "write" requests

https://github.com/Expensify/App/blob/0ed26362d4201ed548525cca7460459d7db6a678/src/libs/API.js#L47-L48

https://github.com/Expensify/App/blob/0ed26362d4201ed548525cca7460459d7db6a678/src/libs/Network/SequentialQueue.js#L90-L96

3. When we come back from offline -> online those queued "write" requests will run

https://github.com/Expensify/App/blob/0ed26362d4201ed548525cca7460459d7db6a678/src/libs/Network/SequentialQueue.js#L83-L84

In this case, we are offline - so shouldn't it be fine to make the request? And when we come back online the report will be fetched?

Is there some problem with the way this is working now that needs to be addressed? Because I want to say it's an anti-pattern to prevent a "write" from happening if we are offline (as that's what the queue itself was designed to do) but maybe I am missing something.

Reply:
The reason I decided to do so was otherwise in the `openReport` method call 
https://github.com/Expensify/App/blob/0bde3f839aa45245df09d9b558eef8b59c066cf1/src/libs/actions/Report.js#L309-L314
it would set `isLoadingReportActions: true` and because of that in the ReportScreen.js 
https://github.com/Expensify/App/blob/0bde3f839aa45245df09d9b558eef8b59c066cf1/src/pages/home/ReportScreen.js#L317-L322
we will keep showing the animating full-screen skeletonUI. This change will prevent that from happening.

But with the recent code change, I can see we are showing the compose box along with the skeletonUI on the report screen page so I can clean all of this.

Reply:
This change and one [below](https://github.com/Expensify/App/pull/13233#discussion_r1059352019) in ReportActionsView page are mainly to prevent setting value for a report prop. Do you have any recommendations on how to handle this better?

Reply:
Let's stop here and consult the team on the best way to handle this. 

The main question in my mind is why we should ever set any loading state before the API request actually begins? In the current design, it seems like we have a hybrid where loading states are set to `true` even though we are offline and not actually "loading" anything.

Taking a look at our usages elsewhere - and this `isLoading` is mostly only being used in a few "blocking read requests" (stuff related to payments and bank accounts) that I don't _think_ will ever happen unless we are online. So it feels like there's an inconsistency or exception being made for the report page - and I'm not sure it's valid.